File: mifosng-android/src/main/java/com/mifos/mifosxdroid/injection/module/ApplicationModule.java
package com.mifos.mifosxdroid.injection.module;

import android.app.Application;
import android.content.Context;

import com.mifos.api.BaseApiManager;
import com.mifos.mifosxdroid.injection.ApplicationContext;
import com.mifos.utils.PrefManager;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import kotlin.Pair;

/**
 * @author Rajan Maurya
 *         Provide application-level dependencies.
 */
@Module
public class ApplicationModule {
    protected final Application mApplication;

    public ApplicationModule(Application application) {
        mApplication = application;
    }

    @Provides
    Application provideApplication() {
        return mApplication;
    }

    @Provides
    @ApplicationContext
    Context provideContext() {
        return mApplication;
    }

    @Provides
    @Singleton
    BaseApiManager provideBaseApiManager() {
        return new BaseApiManager();
    }

    @Provides
    @Singleton
    org.mifos.core.apimanager.BaseApiManager provideSdkBaseApiManager() {
        Pair<String, String> usernamePassword = PrefManager.INSTANCE.getUsernamePassword();
        org.mifos.core.apimanager.BaseApiManager manager = org.mifos.core.apimanager.BaseApiManager
                .Companion.getInstance();
        manager.createService(usernamePassword.getFirst(),
                usernamePassword.getSecond(),
                PrefManager.INSTANCE.getInstanceUrl(),
                PrefManager.INSTANCE.getTenant(), false);
        return manager;
    }

}

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/injection/module/ApplicationModule.kt
package com.mifos.mifosxdroid.injection.module

import android.content.Context
import com.mifos.utils.NetworkUtilsWrapper
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.qualifiers.ApplicationContext
import dagger.hilt.components.SingletonComponent
import javax.inject.Singleton


/**
 * @author Rajan Maurya
 * Provide application-level dependencies.
 */
@Module
@InstallIn(SingletonComponent::class)
class ApplicationModule {

    @Provides
    @Singleton
    fun provideNetworkUtilsWrapper(@ApplicationContext context: Context): NetworkUtilsWrapper {
        return NetworkUtilsWrapper(context)
    }

}